pkg/metadata/labels/types.go

```diff
@@ -15,9 +15,11 @@ var K8SCommon = struct {
 var ODH = struct {
 	OwnedNamespace string
 	Component      func(string) string
+	ComponentName  string
 }{
 	OwnedNamespace: "opendatahub.io/generated-namespace",
 	Component: func(name string) string {
 		return ODHAppPrefix + "/" + name
 	},
+	ComponentName: "component.opendatahub.io/name",
```

Comment:
Is this label already established (i.e., does anything currently depend on it?), or is it a new addition we're considering?

I also see a need for better means of defining taxonomy for ODH. It seems your label somewhat overlaps with `platform.opendatahub.io/component` I put in [my proposal](https://github.com/opendatahub-io/architecture-decision-records/pull/26).

If it's a new addition, perhaps moving forward with [the proposal](https://github.com/opendatahub-io/architecture-decision-records/pull/26) together would be beneficial. 

I think it would be good first to have it fleshed out and agreed on, so we can proceed with enriching all components similarly.

WDYT @dhirajsb?

Reply:
> [!NOTE]
> This file is based on existing (legacy?) labels. The intention was to gather them together so they're easy to discover - a bit more explanation in the related PR #941

Reply:
I'm happy with moving to a newer convention if there is one.

Reply:
But, my understanding is that it would have to be a big refactor across all components to standardize this new label.

Reply:
I might be wrong, but where is `component.opendatahub.io/name` currently in use?

Reply:
Yeah, that label is being referenced in [multiple places](https://github.com/search?q=org%3Aopendatahub-io%20component.opendatahub.io%2Fname&type=code). I'm guessing the dashboard might also be using that to search for resources belonging to a component.